rust/src/dataset/scanner.rs

```diff
@@ -526,8 +532,20 @@ impl Scanner {
                 );
                 // first we do flat search on just the new data
                 let topk_appended = self.flat_knn(scan_node, q)?;
+
+                // To do a union, we need to make the schemas match. Right now
```

Comment:
just to confirm on the schema: if the user specified projection columns, that's handled elsewhere right? That doesn't affect the schema for either results here?

Reply:
Yes, this node always only has these three columns. Later nodes perform `take()` to get the other columns.